src/main/java/bio/terra/cli/service/AxonServerService.java

```diff
@@ -0,0 +1,205 @@
+public class AxonServerService {
+  private static final Logger logger = LoggerFactory.getLogger(AxonServerService.class);
+  // the client object used for talking to Axon Server
+  private final ApiClient apiClient;
+
+  /**
+   * Constructor for class that talks to Axon Server. If the accessToken is null, only
+   * unauthenticated endpoints can be called.
+   */
+  private AxonServerService(@Nullable AccessToken accessToken, Server server) {
+    this.apiClient = new ApiClient();
+
+    this.apiClient.setBasePath(server.getAxonServerUri());
+    if (accessToken != null) {
+      this.apiClient.setAccessToken(accessToken.getTokenValue());
+    }
+  }
+
+  public static AxonServerService fromContext() {
+    return new AxonServerService(Context.requireUser().getTerraToken(), Context.getServer());
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/metadata:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public ClusterMetadata getClusterMetadata(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterMetadata(workspaceId, resourceId),
+        "Failed to retrieve cluster metadata");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/componentUrl:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   * @param componentKey the component key, e.g. "JupyterLab"
+   */
+  public bio.terra.axonserver.model.Url getClusterComponentUrl(
+      UUID workspaceId, UUID resourceId, String componentKey) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterComponentUrl(workspaceId, resourceId, componentKey),
+        "Failed to retrieve cluster component url");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/status:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public ClusterStatus getClusterStatus(UUID workspaceId, UUID resourceId) {
+    GcpResourceApi gcpResourceApi = new GcpResourceApi(apiClient);
+    return callWithRetries(
+        () -> gcpResourceApi.getDataprocClusterStatus(workspaceId, resourceId),
+        "Failed to retrieve cluster status");
+  }
+
+  /**
+   * Call the Axon Server GET
+   * /api/workspaces/v1/{workspaceId}/resources/{resourceId}/gcp/dataproccluster/start:
+   *
+   * @param workspaceId the workspace that contains the resource
+   * @param resourceId the cluster resource id
+   */
+  public void putStartCluster(UUID workspaceId, UUID resourceId) {
```

Comment:
super nit: can they just be called startCluster and stopCluster

Reply:
Done.